### PR TITLE
CalibTracker/SiStripESProducers: ESProducers update to return unique_ptr.

### PR DESCRIPTION
**Pull request by gartung**

### CODE REVIEW

**Comment by ghellwig** [comment]
please test

File: CalibTracker/SiStripESProducers/plugins/fake/SiStripApvGainFakeESSource.cc [hunk 1]
```diff
@@ -71,7 +71,7 @@ SiStripApvGainFakeESSource::produce(const SiStripApvGainRcd& iRecord)
 {
   using namespace edm::es;
 
-  std::shared_ptr<SiStripApvGain> apvGain{new SiStripApvGain};
+  std::unique_ptr<SiStripApvGain> apvGain{new SiStripApvGain};
```
Comment (ghellwig) [inline CalibTracker/SiStripESProducers/plugins/fake/SiStripApvGainFakeESSource.cc]: please change to
```
auto apvGain = std::make_unique<SiStripApvGain>();
```

File: CalibTracker/SiStripESProducers/plugins/fake/SiStripBackPlaneCorrectionFakeESSource.cc [hunk 2]
```diff
@@ -67,7 +67,7 @@ SiStripBackPlaneCorrectioNFakeESSource::produce(const SiStripBackPlaneCorrection
   edm::ESHandle<TrackerTopology> tTopo;
   iRecord.getRecord<TrackerTopologyRcd>().get(tTopo);
 
-  std::shared_ptr<SiStripBackPlaneCorrection> backPlaneCorrection{new SiStripBackPlaneCorrection};
+  std::unique_ptr<SiStripBackPlaneCorrection> backPlaneCorrection = std::make_unique<SiStripBackPlaneCorrection>();
```
Comment (ghellwig) [inline CalibTracker/SiStripESProducers/plugins/fake/SiStripBackPlaneCorrectionFakeESSource.cc]: please use `auto`

File: CalibTracker/SiStripESProducers/plugins/fake/SiStripBadModuleConfigurableFakeESSource.cc [hunk 3]
```diff
@@ -73,7 +73,7 @@ SiStripBadModuleConfigurableFakeESSource::produce(const SiStripBadModuleRcd& iRe
   edm::ESHandle<TrackerTopology> tTopo;
   iRecord.getRecord<TrackerTopologyRcd>().get(tTopo);
 
-  std::shared_ptr<SiStripQuality> quality{new SiStripQuality};
+  std::unique_ptr<SiStripQuality> quality = std::make_unique<SiStripQuality>();
```
Comment (ghellwig) [inline CalibTracker/SiStripESProducers/plugins/fake/SiStripBadModuleConfigurableFakeESSource.cc]: please use `auto`

File: CalibTracker/SiStripESProducers/plugins/fake/SiStripBaseDelayFakeESSource.cc [hunk 4]
```diff
@@ -64,7 +64,7 @@ SiStripBaseDelayFakeESSource::produce(const SiStripBaseDelayRcd& iRecord)
 {
   using namespace edm::es;
 
-  std::shared_ptr<SiStripBaseDelay> baseDelay{new SiStripBaseDelay};
+  std::unique_ptr<SiStripBaseDelay> baseDelay = std::make_unique<SiStripBaseDelay>();
```
Comment (ghellwig) [inline CalibTracker/SiStripESProducers/plugins/fake/SiStripBaseDelayFakeESSource.cc]: please use `auto`

File: CalibTracker/SiStripESProducers/plugins/fake/SiStripConfObjectFakeESSource.cc [hunk 5]
```diff
@@ -61,7 +61,7 @@ SiStripConfObjectFakeESSource::produce(const SiStripConfObjectRcd& iRecord)
 {
   using namespace edm::es;
 
-  std::shared_ptr<SiStripConfObject> confObject{new SiStripConfObject};
+  std::unique_ptr<SiStripConfObject> confObject = std::make_unique<SiStripConfObject>();
```
Comment (ghellwig) [inline CalibTracker/SiStripESProducers/plugins/fake/SiStripConfObjectFakeESSource.cc]: please use `auto`

File: CalibTracker/SiStripESProducers/plugins/fake/SiStripLatencyFakeESSource.cc [hunk 6]
```diff
@@ -64,7 +64,7 @@ SiStripLatencyFakeESSource::produce(const SiStripLatencyRcd& iRecord)
 {
   using namespace edm::es;
 
-  std::shared_ptr<SiStripLatency> latency{new SiStripLatency};
+  std::unique_ptr<SiStripLatency> latency = std::make_unique<SiStripLatency>();
```
Comment (ghellwig) [inline CalibTracker/SiStripESProducers/plugins/fake/SiStripLatencyFakeESSource.cc]: please use `auto`

File: CalibTracker/SiStripESProducers/plugins/fake/SiStripLorentzAngleFakeESSource.cc [hunk 7]
```diff
@@ -169,7 +169,7 @@ SiStripLorentzAngleFakeESSource::produce(const SiStripLorentzAngleRcd& iRecord)
   edm::ESHandle<TrackerTopology> tTopo;
   iRecord.getRecord<TrackerTopologyRcd>().get(tTopo);
 
-  std::shared_ptr<SiStripLorentzAngle> lorentzAngle{new SiStripLorentzAngle};
+  std::unique_ptr<SiStripLorentzAngle> lorentzAngle = std::make_unique<SiStripLorentzAngle>();
```
Comment (ghellwig) [inline CalibTracker/SiStripESProducers/plugins/fake/SiStripLorentzAngleFakeESSource.cc]: please use `auto`

File: CalibTracker/SiStripESProducers/plugins/fake/SiStripNoisesFakeESSource.cc [hunk 8]
```diff
@@ -98,7 +98,7 @@ SiStripNoisesFakeESSource::produce(const SiStripNoisesRcd& iRecord)
   edm::ESHandle<TrackerTopology> tTopo;
   iRecord.getRecord<TrackerTopologyRcd>().get(tTopo);
 
-  std::shared_ptr<SiStripNoises> noises{new SiStripNoises};
+  std::unique_ptr<SiStripNoises> noises = std::make_unique<SiStripNoises>();
```
Comment (ghellwig) [inline CalibTracker/SiStripESProducers/plugins/fake/SiStripNoisesFakeESSource.cc]: please use `auto`

File: CalibTracker/SiStripESProducers/plugins/fake/SiStripPedestalsFakeESSource.cc [hunk 9]
```diff
@@ -64,7 +64,7 @@ SiStripPedestalsFakeESSource::produce(const SiStripPedestalsRcd& iRecord)
 {
   using namespace edm::es;
 
-  std::shared_ptr<SiStripPedestals> pedestals{new SiStripPedestals};
+  std::unique_ptr<SiStripPedestals> pedestals = std::make_unique<SiStripPedestals>();
```
Comment (ghellwig) [inline CalibTracker/SiStripESProducers/plugins/fake/SiStripPedestalsFakeESSource.cc]: please use `auto`

File: CalibTracker/SiStripESProducers/plugins/real/SiStripBadModuleFedErrESSource.cc [hunk 10]
```diff
@@ -125,7 +125,7 @@ SiStripBadModuleFedErrESSource::produce(const SiStripBadModuleFedErrRcd& iRecord
   edm::ESHandle<SiStripFedCabling> cabling;
   iRecord.getRecord<SiStripFedCablingRcd>().get(cabling);
 
-  std::shared_ptr<SiStripQuality> quality{new SiStripQuality};
+  std::unique_ptr<SiStripQuality> quality = std::make_unique<SiStripQuality>();
```
Comment (ghellwig) [inline CalibTracker/SiStripESProducers/plugins/real/SiStripBadModuleFedErrESSource.cc]: please use `auto`

File: CalibTracker/SiStripESProducers/plugins/fake/SiStripThresholdFakeESSource.cc [hunk 11]
```diff
@@ -66,7 +66,7 @@ SiStripThresholdFakeESSource::produce(const SiStripThresholdRcd& iRecord)
 {
   using namespace edm::es;
 
-  std::shared_ptr<SiStripThreshold> threshold{new SiStripThreshold};
+  std::shared_ptr<SiStripThreshold> threshold = std::make_shared<SiStripThreshold>();
```
Comment (ghellwig) [inline CalibTracker/SiStripESProducers/plugins/fake/SiStripThresholdFakeESSource.cc]: The title of this PR is to move to unique_ptr. Can you quickly explain, why this is not extended to all conditions?

**Comment by ghellwig** [comment]
please test

**Comment by ghellwig** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**